infra/src/main/kotlin/fi/fta/geoviite/infra/dataImport/SwitchLibraryDataMigration.kt

```diff
@@ -164,19 +273,37 @@ class V53__add_rr54_4x_switch_structure : BaseJavaMigration() {
     val logger: Logger = LoggerFactory.getLogger(this::class.java)
 
     override fun migrate(context: Context?) {
-        withImportUser(ImportUser.SWITCH_LIB_IMPORT) {
-            logger.info("Add RR54-4x1:9 structure into library")
+        // Switch structures are now migrated repeatably
+    }
+
+    override fun getChecksum(): Int = 1
+}
+
+@Suppress("unused", "ClassName")
+class R__10_01__update_all_switch_structures : BaseJavaMigration() {
+    val logger: Logger = LoggerFactory.getLogger(this::class.java)
 
+    override fun migrate(context: Context?) {
+        withImportUser(ImportUser.SWITCH_LIB_IMPORT) {
+            logger.info("Update all switch structures")
             val connection =
                 context?.connection ?: throw IllegalStateException("Can't run imports without DB connection")
             val jdbcTemplate = NamedParameterJdbcTemplate(
                 SingleConnectionDataSource(connection, true)
             )
             val switchStructureDao = SwitchStructureDao(jdbcTemplate)
-            switchStructureDao.insertSwitchStructure(RR54_4x1_9())
+            val switchOwnerDao = SwitchOwnerDao(jdbcTemplate)
+            val switchLibraryService = SwitchLibraryService(switchStructureDao, switchOwnerDao)
+            switchLibraryService.replaceExistingSwitchStructures(switchStructures)
         }
     }
 
-    // Increase this manually when switch library data changes
-    override fun getChecksum(): Int = 1
+    @OptIn(ExperimentalStdlibApi::class)
+    override fun getChecksum(): Int {
+        val objectMapper = ObjectMapper()
+        val json = objectMapper.writeValueAsString(switchStructures.map { it.stripUniqueIdentifiers() })
+        val md = MessageDigest.getInstance("MD5")
```

Comment:
Flyway:n omat helpperit ei taida tarjota muita kuin resources-pohjaisia checksum-laskentoja, joten ehkä tämä pitää tehdä itse mutta mietin vielä että onkohan tuossa md5:ssä mitään hyötyä kun senkin stringistä otetaan vain hashCode :thinking: 

SwitchStructure on jo puhdas dataluokka, joten sillä on automaattisesti generoitu hashCode() funkkari jo itsessään. Se ei myöskään muutu jos jacksonin serialisointi muuttuu (kuten tämä json-pohjainen tekee). Eli tälle voisi myös tehdä vain näin, enkä heti keksi miksi se ei olisi yhtä uniikki:
```
override fun getChecksum(): Int = switchStructures.hashCode()
```

Dataluokilla on myös generoitu `.toString()`, joten jsonia ei tarvitse tuottaa jacksonin kautta, vaikka md5:stä haluaisikin hyödyntää.

Reply:
Tuo vaikuttaa joo fiksulta, testaan